### ISSUE TITLE
Any way can get GPU work?

### ISSUE THREAD

**Issue by rexzhang2023** [imartinez/privateGPT#59]
Can anyone suggest how to make GPU work with this project?

**Comment by walking-octopus** [comment]
Chances are, it's already partially using the GPU. As it is now, it's a script linking together LLaMa.cpp emeddings, Chroma vector DB, and GPT4All. GPT4All might be using PyTorch with GPU, Chroma is probably already heavily CPU parallelized, and LLaMa.cpp runs only on the CPU.

It's also worth noting that two LLMs are used with different inference implementations, meaning you may have to load the model twice.

**Comment by mmike87** [comment]
I watched my GPU usage and it was not touched.

**Comment by pabl-o-ce** [comment]
this mean that this work only with CPU? 

I currently want to try this

Also can give some info on the Readme about the requirements of hardware.

**Comment by su77ungr** [comment]
No, LlamaCpp was designed to take only CPU resources. For GPU you'd have to use the native Llama model from facebook.

**Comment by iker-lluvia** [comment]
I can get it work installing llama-cpp-python with cuBLAS:
`CMAKE_ARGS="-DLLAMA_CUBLAS=on" FORCE_CMAKE=1 pip install llama-cpp-python==0.1.48`

If installation fails because it doesn't find CUDA, it's probably because you have to include CUDA install path to PATH environment variable:
`export PATH=/usr/local/cuda/bin${PATH:+:${PATH}}`

Anyways, it only uses lesst than 1 GB of the VRAM on a RTX 2060 with 6 GB, so I don't know if something is still missing.

**Comment by su77ungr** [comment]
Aren't you just emulating the CPU?
Idk if there's even working port for GPU support

**Comment by iker-lluvia** [comment]
> Aren't you just emulating the CPU? Idk if there's even working port for GPU support

It shouldn't. The [llama.cpp library](https://github.com/ggerganov/llama.cpp) can perform BLAS acceleration using the CUDA cores of the Nvidia GPU through cuBLAS. I expect llama-cpp-python to do so as well when installing with cuBLAS.
Any fast way to verify if the GPU is being used other than running `nvidia-smi` or `nvtop`?

**Comment by su77ungr** [comment]
Nvm my collaborator found a way [see](https://github.com/hippalectryon-0/CASALIOY/blob/3af2ae5edc1f6c0adfc2fb0cfb3e4ac852b82dbd/Dockerfile)
